src/backend/storage/file/buffile.c

```diff
@@ -998,6 +1000,35 @@ BufFilePledgeSequential(BufFile *buffile)
 
 #define BUFFILE_ZSTD_COMPRESSION_LEVEL 1
 
+typedef struct {
+	ResourceOwner owner;
+} ResourceOwnerContext;
+
+void *
+customAlloc(void* opaque, size_t size)
+{
+	ResourceOwnerContext *resContext = (ResourceOwnerContext*)opaque; 
+	ResourceOwner oldowner = CurrentResourceOwner;
+	CurrentResourceOwner = resContext->owner;
+
+	void *mem = MemoryContextAlloc(TopMemoryContext, size);
```

Comment:
This is all, of course, good, but maybe, to protect against memory leaks, in general, create your own memory context, which is deleted at the end?

Reply:
Seems like that won't be easy as that memory context should be allocated and deleted outside `BufFileStartCompression` and `BufFileEndCompression` and it will require to modify plenty of code